imod/mf6/rch.py

```diff
@@ -166,6 +167,22 @@ def __init__(
         super().__init__(dict_dataset)
         self._validate_init_schemata(validate)
 
+    @classmethod
+    def from_file(cls, path: str | Path, **kwargs) -> Self:
+        instance = super().from_file(path, **kwargs)
+
+        # to_netcdf converts strings into  NetCDF "variable‑length UTF‑8 strings"
+        # which are loaded as dtype=object arrays.
+        # This will convert them back to str.
+        vars = [
+            "species",
+        ]
+        for var in vars:
+            if var in instance.dataset:
+                instance.dataset[var] = instance.dataset[var].astype(str)
+
+        return instance
```

Comment:
I think this should be part of the pkgbase class. Or BoundaryCondition class, as species can also be part of other packages.